docs/types.rst

```diff
@@ -115,6 +115,23 @@ Operator       Description
 
 ``x`` and ``y`` must be of the same type.
 
+Shifts
+^^^^^^^^^^^^^^^^^
+
+=============  ======================
+Operator       Description
+=============  ======================
+``x << y``     Left shift
+``x >> y``     Right shift
+=============  ======================
+
+Shifting is only available for 256-bit wide types. That is, ``x`` must be ``int256``, and ``y`` can be any unsigned integer. The right shift for ``int256`` compiles to a signed right shift (EVM ``SAR`` instruction).
```

Comment:
> I might be confused since I haven't checked yet the implementation details, but x can be uint256 OR int256 (here you say x must be int256 and below you say x must be uint256). What you probably mean (that's how I understand it) is that for logical shifting x must be uint256 whilst for arithmetic shift operations x must be int256.

Not sure if I wrote it as clearly as possible, but the rationale was that the docs were already split by signed and unsigned integers, and so I split the explanations by section. The first section was already discussing signed integers, and the second section was already discussing unsigned integers.

> Furthermore, it might be useful to add to the docs that the right shift x >> y for negative x is rounded down (towards negative infinity). The reason why I mention this is that in Solidity versions <0.5.0 the same expression (which is equivalent to the mathematical expression x / 2**y) was rounded towards zero. So we're sure people don't take wrong assumptions here.
> 
> Eventually, I also think it's important to highlight that the runtime overflow always returns 0 and not anything else. This contradicts the intuition we have from unchecked arithmetic where everything wraps and starts all over again from 0 but can have any uint256 result.

I don't really think this is necessary - I made sure to explain that the shift operator compiles directly to the SHR/SAR instruction, whose behavior is well known (and is also clarified at length in the text of [EIP-145](https://eips.ethereum.org/EIPS/eip-145)).

Reply:
> Not sure if I wrote it as clearly as possible, but the rationale was that the docs were already split by signed and unsigned integers, and so I split the explanations by section. The first section was already discussing signed integers, and the second section was already discussing unsigned integers.

Right, that makes actually sense :)

> I don't really think this is necessary - I made sure to explain that the shift operator compiles directly to the SHR/SAR instruction, whose behavior is well known (and is also clarified at length in the text of [EIP-145](https://eips.ethereum.org/EIPS/eip-145)).

Ok fair enough, but I doubt many folks even know about this EIP lol.